internal/storage/loopback/safe.go

```diff
@@ -0,0 +1,120 @@
+package loopback
+
+import (
+	"context"
+	"errors"
+	"syscall"
+	"time"
+
+	"golang.org/x/sys/unix"
+)
+
+type (
+	open struct {
+		path string
+		mode int
+		perm uint32
+	}
+
```

Comment:
I've never seen this code structure before...

Reply:
This was copied as is from the daemon, I'm not entirely sure we even need the safe ioctl stuff. But me neither before this haha

Reply:
what daemon?

Reply:
teleportd.